Code/Framework/AzCore/Tests/Jobs.cpp

```diff
@@ -1634,24 +1627,113 @@ namespace UnitTest
+
+    private:
+        const AZStd::string m_name;
+        AZStd::binary_semaphore& m_binarySemaphore;
+        AZStd::vector<AZStd::string>& m_namesOfProcessedJobs;
+    };
+    AZStd::atomic<AZ::s32> TestJobWithPriority2::s_numIncompleteJobs = 5;
+
+    class JobPriorityTestFixture2 : public DefaultJobManagerSetupFixture
+    {
+    public:
+        JobPriorityTestFixture2() : DefaultJobManagerSetupFixture(1) // Only 1 worker to serialize job execution
+        {
+        }
+
+        void RunTest()
+        {
+            AZStd::vector<AZStd::string> namesOfProcessedJobs;
+
+            // The queue is empty, and calling 'Start' on a job inserts it into the job queue, but it won't necesarily begin processing immediately.
+            // So we need to set the priority of the first job queued to the highest available, ensuring it will get processed first even if the lone
+            // worker thread active in this test does not inspect the queue until after all the jobs below have been 'started' (this is guaranteed by
+            // the fact that jobs with equal priority values are processed in FIFO order).
+            //
+            // The binary semaphore ensures that this first job does not complete until we've finished queuing all the other jobs.
+            //
+            // All jobs that we start in this test have the 'isAutoDelete' param set to true, so we don't have to store them or clean up.
+            AZStd::binary_semaphore binarySemaphore;
+            (aznew TestJobWithPriority2(Job::Priority::MEDIUM, "FirstJobQueued", m_jobContext, binarySemaphore, namesOfProcessedJobs))->Start();
+
+            // Release the binary semaphore so the first queued job will complete. The rest of the queued jobs should now complete in order of their priority.
+            binarySemaphore.release();
+
+            // Wait until all the jobs have completed. Ideally we would start one last job (with the lowest priority so it is guaranteed to be processed last
+            // even if the jobs started above have yet to complete) and wait until it has completed, but this results in this main thread picking up jobs and
+            // thorwing all the careful scheduling on the one worker thread active in this test out the window (please see Job::StartAndAssistUntilComplete).
+            while (TestJobWithPriority2::s_numIncompleteJobs > 0) {}
+
+            // Verify that the jobs were completed in the expected order.
+            EXPECT_EQ(namesOfProcessedJobs[0], "FirstJobQueued");
+            EXPECT_EQ(namesOfProcessedJobs[1], "CriticalPriority");
+            EXPECT_EQ(namesOfProcessedJobs[2], "HighPriority");
+            EXPECT_EQ(namesOfProcessedJobs[3], "DefaultPriority");
+            EXPECT_EQ(namesOfProcessedJobs[4], "LowPriority");
+        }
+    };
+
+    // NOTE: This test ONLY tests that jobs are dequeued in priority order from a local worker job queue
+    TEST_F(JobPriorityTestFixture2, Test)
+    {
+        RunTest();
+    }
```

Comment:
Can you run this test locally with a repeat count of 1000?
Just to make sure no deadlocks are seen.

If you are using Visual Studio, You can set the Startup Project to AzCore.Tests and modifying the Debugging parameters to append  `--gtest_repeat=1000`

Reply:
No issues reported. I had to make a minor local change to make the test repeatable in the same executable. I'll roll that into a future PR.